### PR TITLE
Fix SHA2Test. Fix #24

### LINKED ISSUE #24: SHA2Test fails for newer Bytes versions

### PR DESCRIPTION
**Pull request by minborg**

### CODE REVIEW

File: src/test/java/net/openhft/chronicle/salt/SHA2Test.java [hunk 1]
```diff
@@ -31,7 +31,9 @@ private static void doTest256(byte[] inputStr, String expectedHex) {
     }
 
     private static void doTest512(String inputStr, String expectedHex) {
-        Bytes<?> input = Bytes.allocateElasticDirect();
+        Bytes<?> input = Bytes.allocateElasticDirect()
+                // Forces replacement of the underlying BytesStore for ""
+                .write(new byte[0]);
```
Comment (alamar) [inline src/test/java/net/openhft/chronicle/salt/SHA2Test.java:36]: Why? Is there a chance our users will also have to do this? Does allocateElasticDirect() return an EmptyByteStore with all of its methods marked as Unsupported?

Comment (minborg) [reply src/test/java/net/openhft/chronicle/salt/SHA2Test.java:36]: The problem is specifically for low-level methods like addressForRead. They should not be used by end-users.

Comment (minborg) [reply src/test/java/net/openhft/chronicle/salt/SHA2Test.java:36]: `allocateElasticDirect()` returns a `Bytes` initially backed by the `EmptyByteStore`.

Comment (alamar) [reply src/test/java/net/openhft/chronicle/salt/SHA2Test.java:36]: I don't see `addressForRead()` in this test - this check should go to the same place where this domestic method is called. Ideally, there should be no domestic API in Bytes and dependencies should only use public API

**Pull request closed without merging**